### PR TITLE
Add `mana` field to rust outputs

### LINKED ISSUE #865: Add `mana` field to outputs - rust
- [x] [Basic](https://github.com/iotaledger/tips-draft/blob/tip41/tips/TIP-0041/tip-0041.md#basic-output)
- [x] [NFT](https://github.com/iotaledger/tips-draft/blob/tip43/tips/TIP-0043/tip-0043.md#nft-output)
- [x] [Account](https://github.com/iotaledger/tips-draft/blob/tip42/tips/TIP-0042/tip-0042.md#account-output)

### PR DESCRIPTION
**Pull request by DaughterOfMars**
# Description of change

Adds the mana field to Basic, Account, and Nft outputs. I didn't see any validation requirements, but if something is needed let me know.

## Links to any relevant issues

Closes #865

### CODE REVIEW

File: sdk/src/types/block/output/nft.rs [hunk 1]
```diff
@@ -79,6 +81,13 @@ impl NftOutputBuilder {
         self
     }
 
+    /// Sets the mana to the provided value. (∩ ͡° ͜ʖ ͡°)⊃━☆ﾟ. *
```
Comment (thibault-martinez) [inline sdk/src/types/block/output/nft.rs]: Cute but no 😆

Comment (DaughterOfMars) [reply sdk/src/types/block/output/nft.rs]: \>:C give me one good reason

Comment (thibault-martinez) [reply sdk/src/types/block/output/nft.rs]: Consistency, we'd need emojis on all setters

Comment (DaughterOfMars) [reply sdk/src/types/block/output/nft.rs]: That can be arranged

File: bindings/core/src/message_interface_old/account_method.rs [hunk 2]
```diff
@@ -48,6 +48,8 @@ pub enum AccountMethod {
     BuildAccountOutput {
         // If not provided, minimum storage deposit will be used
         amount: Option<String>,
+        #[serde(with = "string")]
+        mana: u64,
```
Comment (thibault-martinez) [inline bindings/core/src/message_interface_old/account_method.rs:53]: Why is that different from the amount?

Comment (thibault-martinez) [reply bindings/core/src/message_interface_old/account_method.rs:53]: I mean, why not a String

Comment (DaughterOfMars) [reply bindings/core/src/message_interface_old/account_method.rs:53]: It's better to use the real type and let serde manage the conversions, but I didn't want to also re-implement all my changes from #893 in this PR.

File: sdk/src/types/block/output/nft.rs (hunk 1 above)
Comment (thibault-martinez) [reply sdk/src/types/block/output/nft.rs]: Are you really insisting on keeping the emoji or can we remove it?

File: bindings/core/src/message_interface_old/account_method.rs [hunk 3]
```diff
@@ -64,6 +66,8 @@ pub enum AccountMethod {
     BuildBasicOutput {
         // If not provided, minimum storage deposit will be used
         amount: Option<String>,
+        #[serde(default, with = "string")]
```
Comment (thibault-martinez) [inline bindings/core/src/message_interface_old/account_method.rs:71]: Why is this one default and not the others?

Comment (DaughterOfMars) [reply bindings/core/src/message_interface_old/account_method.rs:71]: Because I forgot lol

File: sdk/tests/client/input_selection/nft_outputs.rs [hunk 4]
```diff
@@ -106,58 +106,60 @@ fn transition_nft_id_zero() {
...
+// }
 
 #[test]
 fn basic_output_with_nft_input() {
     let protocol_parameters = protocol_parameters();
     let nft_id_2 = NftId::from_str(NFT_ID_2).unwrap();
 
     let inputs = build_inputs([Nft(
-        2_229_500,
+        2_233_500,
```
Comment (thibault-martinez) [inline sdk/tests/client/input_selection/nft_outputs.rs:162]:
```suggestion
        2_229_500,
```
Let's keep the original value if we comment

File: sdk/tests/client/input_selection/nft_outputs.rs [hunk 5]
```diff
@@ -282,50 +284,50 @@ fn burn_nft() {
...
+//         addresses([BECH32_ADDRESS_ED25519_0]),
+//         protocol_parameters,
+//     )
+//     .select();
+
+//     assert!(matches!(
+//         selected,
+//         Err(Error::InsufficientAmount {
+//             found: 1_000_001,
+//             required: 1_217_000,
```
Comment (thibault-martinez) [inline sdk/tests/client/input_selection/nft_outputs.rs:327]:
```suggestion
//             required: 1_213_000,
```

File: sdk/tests/client/input_selection/nft_outputs.rs (hunk 4 above)
Comment (thibault-martinez) [reply sdk/tests/client/input_selection/nft_outputs.rs:162]: Hmm you changed pretty much everywhere, annoying

Comment (thibault-martinez) [reply sdk/tests/client/input_selection/nft_outputs.rs:162]: I just don't remember how the input values were selected, may be interesting to have them later to fix them

File: bindings/core/src/message_interface_old/account_method.rs [hunk 6]
```diff
@@ -48,6 +48,8 @@ pub enum AccountMethod {
     BuildAccountOutput {
         // If not provided, minimum storage deposit will be used
         amount: Option<String>,
+        #[serde(default, with = "string")]
```
Comment (thibault-martinez) [inline bindings/core/src/message_interface_old/account_method.rs:52]: I don't know yet if mana will be mandatory to build an output, if it is, we may forget to remove this default. I'd maybe drop a temporary TODO until we know more about mana.

File: sdk/tests/client/input_selection/nft_outputs.rs (hunk 4 above)
Comment (DaughterOfMars) [reply sdk/tests/client/input_selection/nft_outputs.rs:162]: Doesn't really matter does it? Feels like a waste of time to revert this.

File: bindings/core/src/message_interface_old/account_method.rs (hunk 6 above)
Comment (DaughterOfMars) [reply bindings/core/src/message_interface_old/account_method.rs:52]: Okay

**Review by thibault-martinez — APPROVED** [review]

**Review by Thoralf-M — APPROVED** [review]

**Review by Alex6323 — APPROVED** [review]

**Review by qrayven — APPROVED** [review]

**Review by kwek20 — APPROVED** [review]

**Pull request merged**